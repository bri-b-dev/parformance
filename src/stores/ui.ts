import { defineStore } from 'pinia'

export const useUiStore = defineStore('ui', {
  state: () => ({
    // transient per-drill UI flags (not persisted)
    gamblerCollapsed: {} as Record<string, boolean>,
  }),
  getters: {
    isGamblerCollapsed: (state) => (drillId: string) => {
      if (!drillId) return true
      return state.gamblerCollapsed[drillId] ?? true
    }
  },
  actions: {
    setGamblerCollapsed(drillId: string, collapsed: boolean) {
      this.gamblerCollapsed = { ...(this.gamblerCollapsed || {}), [drillId]: collapsed }
    }
  }
})
