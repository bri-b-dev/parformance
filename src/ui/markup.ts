// Lightweight HTML markup generators for visual snapshot testing and reuse in simple components.
// These return plain HTML strings (without runtime dependencies) so tests can snapshot them easily.

export function renderSpinner(label: string = 'Ladenâ€¦'): string {
  return [
    '<div class="flex items-center justify-center p-4" role="status" aria-live="polite" aria-busy="true">',
    '  <span class="inline-block h-5 w-5 mr-2 rounded-full border-2 border-gray-300 border-t-blue-500 animate-spin"></span>',
    `  <span class="text-sm text-gray-600">${escapeHtml(label)}</span>`,
    '</div>'
  ].join('\n')
}

export function renderEmptyState(title: string, message?: string): string {
  const heading = `<h3 class=\"text-base font-semibold text-gray-800\">${escapeHtml(title)}</h3>`
  const body = message
    ? `\n  <p class=\"text-sm text-gray-600 mt-1\">${escapeHtml(message)}</p>`
    : ''
  return [
    '<div class="p-6 text-center border border-dashed rounded-md border-gray-200 bg-gray-50">',
    `  ${heading}${body}`,
    '</div>'
  ].join('\n')
}

function escapeHtml(input: string): string {
  return input
    .replace(/&/g, '&amp;')
    .replace(/</g, '&lt;')
    .replace(/>/g, '&gt;')
    .replace(/"/g, '&quot;')
    .replace(/'/g, '&#39;')
}

export default { renderSpinner, renderEmptyState }
