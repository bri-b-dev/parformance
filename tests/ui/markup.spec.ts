import { describe, it, expect } from 'vitest'
import { renderSpinner, renderEmptyState } from '../../src/ui/markup'

describe('UI markup snapshots', () => {
  it('spinner default label', () => {
    expect(renderSpinner()).toMatchSnapshot()
  })

  it('spinner custom label', () => {
    expect(renderSpinner('Bitte wartenâ€¦')).toMatchSnapshot()
  })

  it('empty state with title only', () => {
    expect(renderEmptyState('Keine Ergebnisse')).toMatchSnapshot()
  })

  it('empty state with title and message', () => {
    const html = renderEmptyState('Keine Drills gefunden', 'Passe die Filter an oder lege ein neues Trainingsspiel an.')
    expect(html).toMatchSnapshot()
  })
})
