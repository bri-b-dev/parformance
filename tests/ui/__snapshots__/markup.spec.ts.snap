// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UI markup snapshots > empty state with title and message 1`] = `
"<div class="p-6 text-center border border-dashed rounded-md border-gray-200 bg-gray-50">
  <h3 class="text-base font-semibold text-gray-800">Keine Drills gefunden</h3>
  <p class="text-sm text-gray-600 mt-1">Passe die Filter an oder lege ein neues Trainingsspiel an.</p>
</div>"
`;

exports[`UI markup snapshots > empty state with title only 1`] = `
"<div class="p-6 text-center border border-dashed rounded-md border-gray-200 bg-gray-50">
  <h3 class="text-base font-semibold text-gray-800">Keine Ergebnisse</h3>
</div>"
`;

exports[`UI markup snapshots > spinner custom label 1`] = `
"<div class="flex items-center justify-center p-4" role="status" aria-live="polite" aria-busy="true">
  <span class="inline-block h-5 w-5 mr-2 rounded-full border-2 border-gray-300 border-t-blue-500 animate-spin"></span>
  <span class="text-sm text-gray-600">Bitte warten…</span>
</div>"
`;

exports[`UI markup snapshots > spinner default label 1`] = `
"<div class="flex items-center justify-center p-4" role="status" aria-live="polite" aria-busy="true">
  <span class="inline-block h-5 w-5 mr-2 rounded-full border-2 border-gray-300 border-t-blue-500 animate-spin"></span>
  <span class="text-sm text-gray-600">Laden…</span>
</div>"
`;
